_overviews/scala3-book/interacting-with-java.md

```diff
@@ -39,209 +39,277 @@ Note that the Java examples in this section assume that you’re using Java 11 o
-  static Optional<String> oEmptyString = Optional.empty();
+{% tabs bar-definition %}
+{% tab Java %}
+```java
+public interface Bar {
+  static java.util.Optional<String> optionalString();
 }
 ```
+{% endtab %}
+{% endtabs %}
 
-Now in your Scala code you can access those fields.
-If you just access them directly, they’ll both be `Optional` values:
+First import the `scala.jdk.OptionConverters` object, and then use the `toScala` method to convert the `Optional` value to a Scala `Option`:
 
+{% tabs bar-usage class=tabs-scala-version %}
+{% tab 'Scala 2' %}
 ```scala
-// scala
 import java.util.Optional
+import scala.jdk.OptionConverters._
 
-val optionalString = JavaClass.oString         // Optional[foo]
-val eOptionalString = JavaClass.oEmptyString   // Optional.empty
+val javaOptString: Optional[String] = Bar.optionalString
+val scalaOptString: Option[String] = javaOptString.toScala
 ```
-
-But by using the _scala.jdk.OptionConverters_ methods, you can convert them to Scala `Option` values:
-
+{% endtab %}
+{% tab 'Scala 3' %}
 ```scala
 import java.util.Optional
 import scala.jdk.OptionConverters.*
 
-val optionalString = JavaClass.oString         // Optional[foo]
-val optionString = optionalString.toScala      // Some(foo)
-
-val eOptionalString = JavaClass.oEmptyString   // Optional.empty
-val eOptionString = eOptionalString.toScala    // None
+val javaOptString: Optional[String] = Bar.optionalString
+val scalaOptString: Option[String] = javaOptString.toScala
 ```
-
-
+{% endtab %}
+{% endtabs %}
 
 ## Extending Java interfaces in Scala
 
 If you need to use Java interfaces in your Scala code, extend them just as though they are Scala traits.
 For example, given these three Java interfaces:
 
+{% tabs animal-definition %}
+{% tab Java %}
 ```java
-// java
-interface Animal {
+public interface Animal {
   void speak();
 }
 
-interface Wagging {
+public interface Wagging {
   void wag();
 }
 
-interface Running {
+public interface Running {
   // an implemented method
   default void run() {
     System.out.println("I’m running");
   }
 }
 ```
+{% endtab %}
+{% endtabs %}
 
 you can create a `Dog` class in Scala just as though you were using traits.
 All you have to do is implement the `speak` and `wag` methods:
 
+{% tabs animal-usage class=tabs-scala-version %}
+{% tab 'Scala 2' %}
 ```scala
-// scala
-class Dog extends Animal, Wagging, Running:
+class Dog extends Animal with Wagging with Running {
   def speak = println("Woof")
   def wag = println("Tail is wagging")
+}
 
-@main def useJavaInterfaceInScala =
-  val d = new Dog
+def useJavaInterfaceInScala = {
+  val d = new Dog()
   d.speak
   d.wag
+  d.run
+}
 ```
+{% endtab %}
+{% tab 'Scala 3' %}
+```scala
+class Dog extends Animal, Wagging, Running:
+  def speak = println("Woof")
+  def wag = println("Tail is wagging")
 
+def useJavaInterfaceInScala =
+  val d = Dog()
+  d.speak
+  d.wag
+  d.run
+```
+{% endtab %}
+{% endtabs %}
```

Comment:
Should we also mention the fact that “nilary” methods in Java can be called both with and without parens (ie, `d.run` and `d.run()` are both valid)?

Reply:
I added an explanation